internal/db/service/database.go

```diff
@@ -0,0 +1,230 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"sync"
+
+	_ "github.com/tursodatabase/go-libsql"
+)
+
+type DatabaseService struct {
+	metadataDB   *sql.DB
+	groupDBs     map[string]*sql.DB
+	groupDBMutex sync.RWMutex
+	dbDir        string
+	logger       *slog.Logger
+}
+
+func NewDatabaseService(ctx context.Context, dbDir string, logger *slog.Logger) (*DatabaseService, error) {
+	// Ensure directory exists
+	if err := os.MkdirAll(dbDir, 0755); err != nil {
+		return nil, fmt.Errorf("failed to create db directory: %w", err)
+	}
+
+	// Initialize metadata database
+	metadataPath := filepath.Join(dbDir, "metadata.db")
+	metadataDB, err := sql.Open("libsql", "file:"+metadataPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open metadata database: %w", err)
+	}
+
+	// Configure metadata database with optimized PRAGMA settings
+	if err := configureConnection(ctx, metadataDB, logger); err != nil {
+		if closeErr := metadataDB.Close(); closeErr != nil {
+			logger.Warn("Failed to close metadata database during error cleanup", "error", closeErr)
+		}
+		return nil, fmt.Errorf("failed to configure metadata database: %w", err)
+	}
+
+	// Create metadata tables
+	if err := createMetadataTables(ctx, metadataDB); err != nil {
+		if closeErr := metadataDB.Close(); closeErr != nil {
+			logger.Warn("Failed to close metadata database during error cleanup", "error", closeErr)
+		}
+		return nil, fmt.Errorf("failed to create metadata tables: %w", err)
+	}
+
+	service := &DatabaseService{
+		metadataDB: metadataDB,
+		groupDBs:   make(map[string]*sql.DB),
+		dbDir:      dbDir,
+		logger:     logger,
+	}
+
+	return service, nil
+}
+
+func (s *DatabaseService) Close() error {
+	s.groupDBMutex.Lock()
+	defer s.groupDBMutex.Unlock()
+
+	// Close all group databases
+	for groupID, db := range s.groupDBs {
+		if err := db.Close(); err != nil {
+			s.logger.Error("Failed to close group database", "group_id", groupID, "error", err)
+		}
+	}
+
+	// Close metadata database
+	if err := s.metadataDB.Close(); err != nil {
+		s.logger.Error("Failed to close metadata database", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+func createMetadataTables(ctx context.Context, db *sql.DB) error {
+	queries := []string{
+		`CREATE TABLE IF NOT EXISTS groups (
```

Comment:
Could we move this back into a separate file? It'd be annoying to have to edit this inline